README.md

```diff
@@ -41,121 +46,187 @@ You can find a version of this docummentation, properly rendered, at https://doc
   ```bash
   brew cask install docker
   ```
-- [Docker for Ubuntu](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce-1)
+
+#### Docker on Ubuntu Linux
+
+Follow instructions at [Docker for Ubuntu Linux](https://docs.docker.com/install/linux/docker-ce/ubuntu/#install-docker-ce-1):
+
   ```bash
   sudo apt-get update
   sudo apt-get install docker-ce
   ```
-- [Docker for Arch Linux](https://wiki.archlinux.org/index.php/Docker#Installation)
+
+#### Docker on Arch Linux
+
+Follow instructions at [Docker for Arch Linux](https://wiki.archlinux.org/index.php/Docker#Installation):
+
   ```bash
   sudo pacman -S docker
   ```
 
-#### 2. Install source{d} Engine
+### 2. Install source{d} Engine
 
 Download the **[latest release](https://github.com/src-d/engine/releases)** for MacOS (Darwin) or Linux.
 
-**MacOS:**
+#### engine on macOS
 
 Double-click on the tar file to extract it.
 
-Open your terminal and move it into your local bin folder to make it executable from anywhere:
+Open your terminal and move the binary to your local bin folder to make it executable from any directory:
 
 ```bash
 sudo mv ~/replace/path/to/engine_darwin_amd64/srcd /usr/local/bin/
 ```
 
-**Linux:**
+#### engine on Linux
+
+Extract the contents of the tar file from your terminal:
 
-Extract the tar file from your terminal:
 ```bash
 tar -xvf ~/replace/path/to/engine_REPLACEVERSION_linux_amd64.tar.gz
 ```
 
-Move it into your local bin folder to be executable from anywhere:
+Move the binary to your local bin folder to be executable from any directory:
+
 ```bash
 sudo mv engine_linux_amd64/srcd /usr/local/bin/
 ```
 
-**Windows:**
+#### engine on Windows
 
-_Windows support is coming soon!_
+*Windows support is coming soon!*
 
+For status updates or if you want to help, you may check the [GitHub issue](https://github.com/src-d/engine/issues/45).
 
-#### 3. Start source{d} Engine with your local repositories
+### 3. Start source{d} Engine with your local repositories
 
 Now it's time to initialize the source{d} engine and provide it with some repositories to analyze:
 
-```
-# Without a path it operates on the local folder,
-# it works with nested folders.
+```bash
+# Without a path the engine operates on the local directory
+# it works with nested/sub-directories
 srcd init
 
 # You can also provide a path
 srcd init /home/user/replace/path/
 ```
-**Note:** Ensure that you initialize source{d} Engine every time you want to process a new repository. Changes in the `init` working directory are not detected automatically.
 
-**Note for MaOS:** Docker for Mac [requires file sharing](https://docs.docker.com/docker-for-mac/troubleshoot/#volume-mounting-requires-file-sharing-for-any-project-directories-outside-of-users) for any path outside of `/Users`.
+**Note:**
+Ensure that you initialize source{d} Engine every time you want to process a new repository.
+Changes in the `init` working directory are not detected automatically.
```

Comment:
This is slightly ambiguous: Changes to the _contents_ of the directory are handled correctly, but changes of the working directory _path_ are not. Is that correct?

Reply:
New repositories added to the working directory will not be detected..

I _think_ changes to the already existing repositores will be detected, e.g. a `git pull`. But I may be overlooking some internals of gitbase, like indexes. @ajnavarro?

Reply:
Changes are not supported yet on gitbase: https://github.com/src-d/gitbase/issues/580

Reply:
I added:
>_Database indexes are not updated automatically when its contents change, so the index should be manually recreated._
Is now correct @ajnavarro ?

Reply:
Related to this, another thing that was not clear to me is that running `srcd init` again completely replaces the state of the Engine instance. This is what you want when you had an instance pointing to a repository that changed, but surprised me when I tried to "add" another repository in a different directory.

Reply:
I'm not sure of understanding you @creachadair 

https://github.com/src-d/engine/pull/225#discussion_r271750463
>_[...] another thing that was not clear to me is that running srcd init again completely replaces the state of the Engine instance._

It does. When you run `init` while Engine was already running, it logs:
```
INFO[0000] removing container srcd-cli-bblfshd          
INFO[0000] removing container srcd-cli-daemon           
INFO[0001] removing container srcd-cli-gitbase          
INFO[0001] starting daemon with working directory: /projects/src/github.com/src-d/engine 
```
so all containers are removed and then restored with the new working dir and config.

>_[...] This is what you want when you had an instance pointing to a repository that changed, but surprised me when I tried to "add" another repository in a different directory._

It's a limitation of gitbase: as @ajnavarro said, it does not watch for changes (what is partially true, because it listens for new commits, branches... of the listened to repos, but not for new repos nor updates indexes on new commits.

<hr />

**question:**
Would be this :point_down:  more clear in your opinion? or is also misleading or not clear?

>_**Note:** Once Engine is initialized with a working dir, it does not watch for new repository creation, so if you want to add (or delete) repositories, you need to `init` again._
>_Also, database indexes are not updated automatically when its contents change, so in those cases, the index must be manually recreated._
>_If you want to inspect different datasets, it needs to be done separately: you can `init` one working directory, perform your queries, and then change to the other dataset performing an `init` with the working directory containing it. In such situations, indexes are kept, so you can change from one dataset to the other, without having to recreate them._

The last paragraph (from _"If you want to inspect different datasets [...]"_) might be "too much", so we could omit it if you feel it goes too deep (but I see it's a reasonable and expected use case for engine)

Reply:
@creachadair I just merged the PR but we left this issue opened. If my previous answer does not answer your question... or the current state is not clear enough, or in case you think that my last proposal could be better than the merged code, please let me know in order to clarify it better.
Your suggestions are pretty useful because not being an Engine developer you're more close to the user and gives us an invaluable point of view.

Reply:
I think the paragraph as updated addresses my comments.

I found it a little strange that you can only have one instance running at a time, given that the whole thing is containerized atop managed volumes anyway. As you said, the log messages make it clear what's happened when you re-run `srcd init`…but it seemed counterintuitive to me that the engine couldn't manage more than one starting directory, even aside from the limitation on updates (which makes sense).

But that's not directly related to this change, so I think it was sensible to merge with what you have.

Reply:
I think you're right, and it seems a very reasonable feature request that @src-d/product could consider: **being able to init two working dirs.**

But at this moment, I think it was part of its initial design, maybe (I'm not sure about it) based on the premise that source{d} Engine is meant to analyze a huge dataset (like PGA) so you'd create a dedicated environment for each dataset, or to containing for example all the code of your company.